### PR TITLE
release-template.md: Update for clarity

### PR DESCRIPTION
**Pull request by acicovic**
## Description
This PR updates wording in `release-template.md`, after some suggestions by CodeRabbit.

## Motivation and context
Make the instructions as clear as possible.

## How has this been tested?
No functionality changes.

## Summary by CodeRabbit

- **Documentation**
	- Refined the release tracking instructions to better align with WordPress VIP release schedules.

### CODE REVIEW

**Comment by acicovic** [comment]
@coderabitai review

**Comment by vaurdan** [comment]
@coderabbitai review

**Review by vaurdan — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/release-template.md [hunk 1]
```diff
@@ -6,15 +6,15 @@ labels: 'Type: Maintenance'
...
 
 ## Before releasing
 
 **1. Merge all outstanding work**
 - [ ] Merge any outstanding PRs due for this release to the target branch (usually `develop`).
-- [ ] Verify that all important PRs have an appropriate `Changelog` tag. Those who don't won't be added to the changelog.
+- [ ] Verify that all important PRs have an appropriate `Changelog` tag. PRs without a `Changelog` tag won't be added to the changelog.
 
 **2. Conduct additional testing**
 We've got automated testing in place and also test under our local development environment during development. For impactful releases we should also:
```
[the remark this answers is not in the document]
Comment (acicovic) [reply .github/ISSUE_TEMPLATE/release-template.md:20]: The second note is incorrect. wordpress.org is a URL and does not need capitalization.

[the remark this answers is not in the document]
Comment (acicovic) [reply .github/ISSUE_TEMPLATE/release-template.md:20]: The first note is incorrect. It seems to me that all our bullet points start with capital letters.

**Comment by acicovic** [comment]
@coderabbitai resolve

**Pull request merged**